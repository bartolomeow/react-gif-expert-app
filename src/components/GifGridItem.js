import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ( props ) => {

  return (
      <div className="card animate__animated animate__fadeIn">
        <img src={ props.url } alt={ props.title } />
        <p>{props.title}</p>
      </div>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}

