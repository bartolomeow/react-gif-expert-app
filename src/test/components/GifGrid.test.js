import { shallow } from "enzyme";
import React from "react";
import '@testing-library/jest-dom';
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('Tests for GifGrid component', () => {
    const category = 'One Punch';
 
    test('should show the correct component and match with the snapshot', () => {
        
        useFetchGifs.mockReturnValue({data: [], loading: true});
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot(); 
    });

    test("Debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs", () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier.cosa.jpg',
            title: 'Cualquier cosa'
        },
        {
            id: '123',
            url: 'https://localhost/cualquier.cosa.jpg',
            title: 'Cualquier cosa'
        }];
        useFetchGifs.mockReturnValue({data: gifs, loading: false});
        const wrapper = shallow(<GifGrid category={category} />);

        const GifGridItemLength = wrapper.find('GifGridItem').length;
        const gifslength = gifs.length;

        expect(wrapper).toMatchSnapshot(); 
        expect(wrapper.find('p').exists()).toBe(false);
        expect(GifGridItemLength).toBe(gifslength);
    });
});

