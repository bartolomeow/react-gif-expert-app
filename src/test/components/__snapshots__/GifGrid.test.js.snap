// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GifGrid component Debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3>
     
    One Punch
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier.cosa.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquier cosa"
      url="https://localhost/cualquier.cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Tests for GifGrid component should show the correct component and match with the snapshot 1`] = `
<Fragment>
  <h3>
     
    One Punch
     
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
